pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -115,46 +115,60 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 	}
 
 	var bindingResources []smith_v1.Resource
-	var bindingResult []compute.BindingResult
+	var resourcesWithEnvVarBindings []compute.ResourceWithEnvVarBinding
+	var resourcesWithIamAccessibleBindings []iam.ResourceWithIamAccessibleBinding
 	var references []smith_v1.Reference
 
 	for _, dependency := range dependencies {
-		bindableShape, found, err := knownshapes.FindBindableEnvironmentVariablesShape(dependency.Contract.Shapes)
+		bindableEnvVarShape, envVarFound, err := knownshapes.FindBindableEnvironmentVariablesShape(dependency.Contract.Shapes)
 		if err != nil {
 			return nil, false, err
 		}
-		if !found {
+		if !envVarFound {
 			return nil, false, errors.Errorf("cannot depend on resource %q, only dependencies providing shape %q are supported", dependency.Name, knownshapes.BindableEnvironmentVariablesShape)
 		}
 
-		resourceReference := bindableShape.Data.ServiceInstanceName
+		resourceReference := bindableEnvVarShape.Data.ServiceInstanceName
 		bindingResource := wiringutil.ConsumerProducerServiceBinding(stateResource.Name, dependency.Name, resourceReference)
 		bindingResources = append(bindingResources, bindingResource)
-		bindingResult = append(bindingResult, compute.BindingResult{
+		resourcesWithEnvVarBindings = append(resourcesWithEnvVarBindings, compute.ResourceWithEnvVarBinding{
 			ResourceName:            dependency.Name,
-			BindableEnvVarShape:     *bindableShape,
+			BindableEnvVarShape:     *bindableEnvVarShape,
 			CreatedBindingFromShape: bindingResource,
 		})
-	}
 
-	// filter out the list of dependencyReferences - this will be used
-	dependencyReferences := make([]smith_v1.Reference, 0, len(bindingResult))
-	for _, res := range bindingResult {
-		dependencyReferences = append(dependencyReferences, smith_v1.Reference{
-			Resource: res.CreatedBindingFromShape.Name,
-		})
+		// We also depend on BindableIamAccessible shape
+		bindableIamAccessibleShape, iamFound, err := knownshapes.FindBindableIamAccessibleShape(dependency.Contract.Shapes)
+		if err != nil {
+			return nil, false, err
+		}
+		if iamFound {
```

Comment:
I left this as `if iamFound` such that later if we have more shapes it's a lot easier to understand what is happening.

```
_, found, _ := find()
if found {
   xxx
}

_, found, _ := findAnotherShape()
if !found {
...
}
```

E.g. something like this might happen in the future and it seems more consistent if everything checks a particular branch

Reply:
Then again I don't really care too much so I'll just change it

Reply:
I changed it just to test and tbh it looks a lot messier outside of the if-statement

While there is indenting I think the fact that the code is surrounded by braces makes it easier to see what code belongs to which section

Reply:
Yeah, I see your point. Keep it as is, I agree.